### DISCUSSION TITLE
SIP-4: P2P Specification

### DISCUSSION THREAD

**Discussion by ryanio** [ProjectOpenSea/SIPs#2] [SIP Discussions]
Please use this thread to discuss [SIP-4: P2P Specification](https://github.com/ProjectOpenSea/SIPs/blob/main/SIPS/sip-4.md)

**Comment by Evalir** [comment]
Hey! I've been working along on a Seaport Gossip implementation on [Quay](https://github.com/Alcibiades-Capital/quay). I have a couple of questions regarding the spec:

- For the Kad DHT usage: apart from peer discovery, how does content routing happen here? should orders received / loaded be added to the DHT? In general, the behavior here is still a tad confusing to me—I might be missing something here though, as I'm not very acquainted with the Kademlia dht protocol in general (and i think the ref impl still hasn't finished integrating it).
- As for the wire protocol, it's clear that a `Status` message must be sent on connection with a new peer, but, is the intention that the other messages will be exposed through some API so that they can be used (IPFS's [kubo `cli`](https://docs.ipfs.tech/reference/kubo/cli/#ipfs-dht) comes to mind, even though it could also just be something else)? As they're not explicitly invoked at some other point on the protocol lifecycle.

Also, on the EIP document itself, I noticed some of the wire protocol messages (`GetOrders`, `GetOrderHashes` and `Criteria`) do not  have their matching name on the code type definitions themselves (e.g `GetOrders` has `OrderHashes`)—I think this gets a tad confusing when reading the 6 different messages at first. I think the code examples matching the name would be better for readability, even if the naming is ultimately the responsibility of implementations —if you agree I could send in a PR to change it :).

**Reply by ryanio** [reply]
Hi Evalir! Sorry for the delayed reply, I haven't had much time to keep working on gossip. Definitely open to your contributions to keep improving/working on the spec, can agree that it is still WIP and missing some components. Exciting to see you working on this, and looking forward to collaborating together.
